### PR TITLE
Add builder API response version checking

### PR DESCRIPTION
**Pull request by tbenr**
Makes version field actually parsed and verified when calling builder APIs.

It is actually important in future milestone where we could have no changes in the Payload. So we can early detect an inconsistency with builder milestone and fallback to locally immediately

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Review by StefanBratanov — APPROVED** [review]
LGTM

File: ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/schema/Response.java [hunk 1]
```diff
@@ -41,15 +45,39 @@ public static <T> Response<T> withErrorMessage(final String errorMessage) {
...
+    final boolean milestonesAreMatching = receivedMilestone.equals(expectedMilestone);
+
+    if (strictVersionCheck && !milestonesAreMatching) {
+      throw new IllegalArgumentException(
+          "Wrong response version: expected "
+              + expectedMilestone
+              + ", received "
+              + receivedMilestone);
+    } else {
+      LOG.warn(
```
Comment (StefanBratanov) [inline ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/schema/Response.java]: Isn't this logic always going to log warning, regardless if milestones are matching or not for `getPayload`?

Comment (tbenr) [reply ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/schema/Response.java]: 🤦

**Review by StefanBratanov — APPROVED** [review]
LGTM

**Pull request merged**